### DISCUSSION TITLE
How to set default aggrid font size to smaller for display more content?

### DISCUSSION THREAD

**Discussion by italanchan** [zauberzeug/nicegui#1749] [Q&A]
We want change or set default the font size of aggrid for display more text content without scroll by user interactive,

How to do,?

**Comment by falkoschindler** [comment]
You can change the style of `ui.aggrid`'s default theme "balham" with some custom CSS definitions like this:

```py
ui.add_body_html('''
<style>
.ag-theme-balham {
    --ag-foreground-color: rgb(126, 46, 132);
    --ag-background-color: rgb(249, 245, 227);
    --ag-header-foreground-color: rgb(204, 245, 172);
    --ag-header-background-color: rgb(209, 64, 129);
    --ag-odd-row-background-color: rgb(0, 0, 0, 0.03);
    --ag-header-column-resize-handle-color: rgb(126, 46, 132);

    --ag-font-size: 17px;
    --ag-font-family: monospace;
}
</style>
''')

ui.aggrid({
    'columnDefs': [{'field': 'name'}, {'field': 'age'}],
    'rowData': [{'name': 'John', 'age': 28}, {'name': 'Jane', 'age': 32}],
})
```

<img width="489" alt="Screenshot 2023-10-05 at 21 00 15" src="https://github.com/zauberzeug/nicegui/assets/5767091/4d1dd018-42cb-49f8-8a15-4216ba473365">


The example is adopted from https://www.ag-grid.com/javascript-data-grid/global-style-customisation-colours/ where you can find more information about AG Grid's customization options.

**Reply by mwarrens63** [reply]
Thank you for the rapid response! This is exactly the example for which I have been searching.  By the way, I LOVE the nicegui framework.  I am using nicegui to develop an on-line application for administering a Stapleford-Scored Golf League.  Every screen and several screen components are  written as a python classes.  The MVC design uses nicegui as the View and peewee as the Model.  The Controller code connects the View and Model and implements the "business" logic.  I will be posting the code on github in a couple of  months ...  Thanks again!

**Comment by wikiselev** [comment]
> You can change the style of `ui.aggrid`'s default theme "balham" with some custom CSS definitions like this:
> 
> ```python
> ui.add_body_html('''
> <style>
> .ag-theme-balham {
>     --ag-foreground-color: rgb(126, 46, 132);
>     --ag-background-color: rgb(249, 245, 227);
>     --ag-header-foreground-color: rgb(204, 245, 172);
>     --ag-header-background-color: rgb(209, 64, 129);
>     --ag-odd-row-background-color: rgb(0, 0, 0, 0.03);
>     --ag-header-column-resize-handle-color: rgb(126, 46, 132);
> 
>     --ag-font-size: 17px;
>     --ag-font-family: monospace;
> }
> </style>
> ''')
> 
> ui.aggrid({
>     'columnDefs': [{'field': 'name'}, {'field': 'age'}],
>     'rowData': [{'name': 'John', 'age': 28}, {'name': 'Jane', 'age': 32}],
> })
> ```
> 
> <img alt="Screenshot 2023-10-05 at 21 00 15" width="489" src="https://private-user-images.githubusercontent.com/5767091/273013234-4d1dd018-42cb-49f8-8a15-4216ba473365.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgxMDM1MjgsIm5iZiI6MTcwODEwMzIyOCwicGF0aCI6Ii81NzY3MDkxLzI3MzAxMzIzNC00ZDFkZDAxOC00MmNiLTQ5ZjgtOGExNS00MjE2YmE0NzMzNjUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMTZUMTcwNzA4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDljMGZmZjk5ZmZjZTczMWQyODdlNjI1MWM3MWIxZjNlZDEwNDllNTcxMWE2YjcxMzNmZWYxOTRlYWUxZDBlNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.9aCh8Eso1j60NY-ERB5cCvBo7XkCAwslCmrI1QLo_uY">
> The example is adopted from https://www.ag-grid.com/javascript-data-grid/global-style-customisation-colours/ where you can find more information about AG Grid's customization options.

I am trying to change `balham` theme to `quartz` but it looks like it's not rendering well, there are some missing elements. All other themes seem to work OK.
